tf2/include/tf2/impl/convert.h

```diff
@@ -55,21 +55,33 @@ template <>
 template <typename A, typename B>
 inline void Converter<true, false>::convert(const A& a, B& b)
 {
+#ifdef _MSC_VER
+  tf2::fromMsg(a, b);
+#else
   fromMsg(a, b);
+#endif
 }
 
 template <>
 template <typename A, typename B>
 inline void Converter<false, true>::convert(const A& a, B& b)
 {
+#ifdef _MSC_VER
+  b = tf2::toMsg(a);
+#else
   b = toMsg(a);
+#endif
 }
 
 template <>
 template <typename A, typename B>
 inline void Converter<false, false>::convert(const A& a, B& b)
 {
+#ifdef _MSC_VER
+  tf2::fromMsg(tf2::toMsg(a), b);
```

Comment:
Is it not safe to use this more qualified name ouside of MSVC so we could save the ifdef?

Reply:
As I recalled, it caused compilation errors in GCC\Clang settings but let me test and spin a CI to make sure.

Reply:
@tfoote Here is the error that if I use this full qualified name. It seems that GCC is okay with a unresolved (yet) function name referenced in the template code, but doesn't like the case where you reference it in the fully qualified form.

```
15:39:27 [  9%] Building CXX object CMakeFiles/tf2_ros.dir/src/buffer.cpp.o
15:39:31 In file included from /tmp/ws/install_isolated/include/tf2/convert.h:39:0,
15:39:31                  from /tmp/ws/src/geometry2/tf2_ros/include/tf2_ros/buffer_interface.h:40,
15:39:31                  from /tmp/ws/src/geometry2/tf2_ros/include/tf2_ros/buffer.h:35,
15:39:31                  from /tmp/ws/src/geometry2/tf2_ros/src/buffer.cpp:33:
15:39:31 /tmp/ws/install_isolated/include/tf2/impl/convert.h: In static member function ‘static void tf2::impl::Converter<IS_MESSAGE_A, IS_MESSAGE_B>::convert(const A&, B&) [with A = A; B = B; bool IS_MESSAGE_A = true; bool IS_MESSAGE_B = false]’:
15:39:31 /tmp/ws/install_isolated/include/tf2/impl/convert.h:58:8: error: ‘fromMsg’ is not a member of ‘tf2’
15:39:31    tf2::fromMsg(a, b);
15:39:31         ^~~~~~~
15:39:31 /tmp/ws/install_isolated/include/tf2/impl/convert.h: In static member function ‘static void tf2::impl::Converter<IS_MESSAGE_A, IS_MESSAGE_B>::convert(const A&, B&) [with A = A; B = B; bool IS_MESSAGE_A = false; bool IS_MESSAGE_B = true]’:
15:39:31 /tmp/ws/install_isolated/include/tf2/impl/convert.h:65:12: error: ‘toMsg’ is not a member of ‘tf2’
15:39:31    b = tf2::toMsg(a);
15:39:31             ^~~~~
15:39:31 /tmp/ws/install_isolated/include/tf2/impl/convert.h: In static member function ‘static void tf2::impl::Converter<IS_MESSAGE_A, IS_MESSAGE_B>::convert(const A&, B&) [with A = A; B = B; bool IS_MESSAGE_A = false; bool IS_MESSAGE_B = false]’:
15:39:31 /tmp/ws/install_isolated/include/tf2/impl/convert.h:72:8: error: ‘fromMsg’ is not a member of ‘tf2’
15:39:31    tf2::fromMsg(tf2::toMsg(a), b);
15:39:31         ^~~~~~~
15:39:31 /tmp/ws/install_isolated/include/tf2/impl/convert.h:72:21: error: ‘toMsg’ is not a member of ‘tf2’
15:39:31    tf2::fromMsg(tf2::toMsg(a), b);
15:39:31                      ^~~~~
15:39:31 CMakeFiles/tf2_ros.dir/build.make:62: recipe for target 'CMakeFiles/tf2_ros.dir/src/buffer.cpp.o' failed
15:39:31 make[2]: *** [CMakeFiles/tf2_ros.dir/src/buffer.cpp.o] Error 1
15:39:31 CMakeFiles/Makefile2:248: recipe for target 'CMakeFiles/tf2_ros.dir/all' failed
15:39:31 make[1]: *** [CMakeFiles/tf2_ros.dir/all] Error 2
15:39:31 Makefile:140: recipe for target 'all' failed
15:39:31 make: *** [all] Error 2
15:39:31 <== Failed to process package 'tf2_ros': 
```